File: app/src/main/java/com/example/wsac_app/Parser.java
package com.example.wsac_app;

public class Parser {

    public Parser() {
        //default constructor
    }

    public String getTitle(String title) {
        return title;
    }

    public int getTime(String time) {
        int t;
        try {
            t = Integer.parseInt(time);
        }
        catch(NumberFormatException e) {
            t = -1;
        }
        return t;
    }

    //TODO: Handle Number format exceptions for any Doubles or Ints
    public double getCost(String cost) {
        double c;
        try {
            c = Double.parseDouble(cost);
        }
        catch(NumberFormatException e) {
            c = -1.00;
        }
        return c;
    }

    public int getCal(String cal) {
        int l;
        try {
            l = Integer.parseInt(cal);
        }
        catch(NumberFormatException e) {
            l = -1;
        }
        return l;
    }

    public String[] getIngredients(String ingredients) {
        String[] ingreds = ingredients.split("\n");
        return ingreds;
    }

    public String[] getInstructions(String instructions) {
        String[] instrs = instructions.split("\n");
        return instrs;
    }

    //TODO: parse photo to map to photo id integer in a dictionary or hashmap (separate class)
    public int getPhoto(String photo) {
        return 0;
    }

}

File: app/src/main/java/com/example/wsac_app/Parser.kt
package com.example.wsac_app

import java.lang.NumberFormatException

class Parser {
    fun getTitle(title: String): String {
        return title
    }

    fun getTime(time: String): Int {
        val t: Int
        t = try {
            time.toInt()
        } catch (e: NumberFormatException) {
            -1
        }
        return t
    }

    //TODO: Handle Number format exceptions for any Doubles or Ints
    fun getCost(cost: String): Double {
        val c: Double
        c = try {
            cost.toDouble()
        } catch (e: NumberFormatException) {
            -1.00
        }
        return c
    }

    fun getCal(cal: String): Int {
        val l: Int
        l = try {
            cal.toInt()
        } catch (e: NumberFormatException) {
            -1
        }
        return l
    }

    fun getIngredients(ingredients: String): Array<String> {
        return ingredients.split("\n".toRegex()).toTypedArray()
    }

    fun getInstructions(instructions: String): Array<String> {
        return instructions.split("\n".toRegex()).toTypedArray()
    }

    //TODO: parse photo to map to photo id integer in a dictionary or hashmap (separate class)
    fun getPhoto(photo: String?): Int {
        return 0
    }
}